scripts/generate_ids.py

```diff
@@ -0,0 +1,69 @@
+"""
+ Copyright (C) 2023 Intel Corporation
+ SPDX-License-Identifier: MIT
+ Generates a unique id for each spec function that doesn't have it.
+"""
+
+from fileinput import FileInput
+import util
+import yaml
+
+MAX_FUNCS = 1024 # We could go up to UINT32_MAX...
+ENUM_NAME = '$x_function_t'
+valid_ids = set(range(1, MAX_FUNCS))
+
+class quoted(str):
+    pass
+
+def quoted_presenter(dumper, data):
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='"')
+
+def generate_registry(path, specs):
+    functions = list()
+    etors = dict()
+
+    for s in specs:
+        for i, obj in enumerate(s['objects']):
+            if obj['type'] == 'function':
+                print(obj['class'].removeprefix('$x') + obj['name'])
+                functions.append(obj['class'].removeprefix('$x') + obj['name'])
+
+    try:
+        existing_registry = list(util.yamlRead(path))[1]['etors']
+        for etor in existing_registry:
+            valid_ids.discard(etor['id'])
+            etors[etor['name']] = etor['id']
+    except:
+        print('invalid existing registry, ignoring...')
+        existing_registry = dict()
+
+    updated = False
+
+    new_registry = list()
+    for fname in functions:
+        etor_name = util.to_snake_case(fname)
+        id = etors.get(etor_name)
+        if id is None:
+            updated = True
+            id = valid_ids.pop()
+        new_registry.append({'name': util.to_snake_case(fname), 'desc': 'Enumerator for $x'+fname, 'id': id})
```

Comment:
What happens if a function is removed from the spec?

I think it would no longer appear in the registry but the ID would not be used again but I'm not sure. Does that sound correct?

Reply:
Yup. The id will become available the next time around. I considered this and concluded that addressing this is not worth complicating the script (I'd have to reverse map the etors to function names in the spec).